stable/database/templates/_helpers.tpl

```diff
@@ -556,4 +557,57 @@ the TE Deployment is disabled if TPSG is enabled and this is a secondary release
   {{- else -}}
     {{- include "defaulttrue" .Values.database.te.enablePod -}}
   {{- end -}}
-{{- end -}}
\ No newline at end of file
+{{- end -}}
+
+{{/*
+Renders an ephemeral volume for an engine process.
+*/}}
+{{- define "database.ephemeralVolume" -}}
+{{- $ := index . 0 -}}
+{{- $engine := index . 1 -}}
+{{- if eq (include "defaultfalse" $.Values.database.ephemeralVolume.enabled) "true" }}
+ephemeral:
+  volumeClaimTemplate:
+    metadata:
+      labels:
+        {{- include "database.resourceLabels" $ | nindent 10 }}
+    spec:
+      accessModes:
+      - ReadWriteOnce
+      {{- if $.Values.database.ephemeralVolume.storageClass }}
+      {{- if (eq "-" $.Values.database.ephemeralVolume.storageClass) }}
+      storageClassName: ""
+      {{- else }}
+      storageClassName: {{ $.Values.database.ephemeralVolume.storageClass }}
+      {{- end }}
+      {{- end }}
+      resources:
+        requests:
+          {{- if eq (include "defaultfalse" $.Values.database.ephemeralVolume.sizeToMemory) "true" }}
+          storage: {{ $engine.resources.limits.memory }}
+          {{- else }}
+          storage: {{ $.Values.database.ephemeralVolume.size }}
+          {{- end }}
+{{- else }}
+emptyDir: {}
+{{- end }}
+{{- end -}}
+
+{{/*
+Returns true or false based on whether the ephemeral or emptyDir volume should be rendered.
```

Comment:
Update the comment here as well to make it clear that we can create generic ephemeral volume or emptyDir.